server/aesdsocket.c

```diff
@@ -0,0 +1,216 @@
+#include <stdio.h>
+#include <strings.h>
+#include <unistd.h> 
+#include <sys/socket.h>
+#include <syslog.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <error.h>
+#include <signal.h>
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <string.h>
+
+void sig_handler(int signal)
+{
+	if(unlink("/var/tmp/aesdsocketdata")==-1)
+		perror("unlink failed");
+		
+	syslog(LOG_INFO,"Caught signal,exiting");
```

Comment:
@lnxblog is it safe to call this async-signal unsafe function inside signal handler?